# Mosquito CNV pipeline specification

* Version: 1.0.0
* Authors: Alistair Miles, Eric Lucas, Kevin Palis, Jessica Way

This document specifies a pipeline for calling Copy Number Variants (CNVs) for a
cohort of multiple samples. The pipeline inclludes an HMM step followed by coverage calls and target regions pipelines to improve acuracy. 


## Inputs

* A manifest specifying a set of samples to be called.

* For each sample, a BAM file containing analysis-ready sequence read
  alignments, produced by the [mosquito short read alignment
  pipeline](short-read-alignment-vector.md).

* A  unique sample set ID

* Reference genome sequence (FASTA format).


## Outputs

* CNV HMM output

* CNV Coverage Calls output

* CNV targeted regions output


## Software

* R version 3.6.1


## Pipeline

### Sub-pipeline: CNV HMM 

#### Step: Windowed Coverage
**Description:** Divides the genome into 300bp windows and counts the number of reads for each window.\
**Inputs:** Bam file, 1 per sample\
**Outputs:** Windowed count reads file, 1 per sample \
**Software:** Python virtualenv cnv37\
Steps in CNV_pipeline/scripts/get_windowed_coverage_and_diagnostic_reads.sh\
```bash

```

#### Step: Calculate Coverage Summary Stats
**Description:** Analyzes the read count for each sample. Computes a GC normalization table that describes the mean read count for each percentile of GC. Computs overall coverage variace \
**Inputs:** Windowed count reads file, 1 per sample\
**Outputs:** Summary statistics file, 1 per sample\
**Software:** Python virtualenv cnv37\
Steps in CNV_pipeline/scripts/get_coverage_stats_by_sample_set_vobs.sh\
```bash

```

#### Step: CNV HMM
**Description:** Normalizes the read count for each sample by dividing by the modal coverage for each windw. Fits an HM to the normalized read counts , \
**Inputs:** Windowed count reads file, 1 per sample. Coverage variance statistics, 1 per sample\
**Outputs:** Normalized coverage values, 1 per window, per sample. Copy number state, 1 per window, per sample. \
**Software:** Python virtualenv cnv37\
Steps in CNV_pipeline/scripts/coverage_HMM_vobs.sh\
```bash

```


### Sub-pipeline: CNV Coverage Calls

#### Step: CNV Coverage Calls
**Description:** Uses the predicted copy number state values generated by the HMM to call CNVs. Calls CNVs based on the change in predicted copy number within consecutive windows. Calls variants by grouping CNVs observed in multiple samples, as determined by the start and end coordinates. \
**Inputs:** Copy number state files. 1 per window per sample.\
**Outputs:** File similar to VCF. Lists variants with genotypes for each sample. 1 per sample set.\
**Software:** R version 3.6.1\
Steps in CNV_pipeline/scripts/coverage_CNVs_vobs.sh\
```bash

```


### Sub-pipeline: CNV Targeted Regions 

#### Step: Extract Diagnostic Reads
Note: this is currently combined with the windowed coverage step at the beginning of the pipeline\
**Description:** Extract diagnostic reads needed to call the CNVs in a catalog of CNVs known to be important for resistance. \
**Inputs:** Bam files, 1 per sample. \
**Outputs:** \
**Software:** Python virtualenv cnv37\
Steps in CNV_pipeline/scripts/get_windowed_coverage_and_diagnostic_reads.sh\
```bash

```

#### Step: Target Regions CNV Calls
**Description:** \
**Inputs:** \
**Outputs:** \
**Software:** R version 3.6.1\
Steps in CNV_pipeline/scripts/target_regions_analysis_vobs.sh\
```bash

```

## Implementation notes



## Change log

